Global/EnemyAction.gd

```diff
@@ -2,28 +2,28 @@ extends RefCounted
 
 class_name EnemyAction
 
-var caster: Entity
-var action: CardBase
-var target: Array[Entity]
+var _caster: Entity
+var _action: CardBase
+var _target_list: Array[Entity]
 	
-func _init(caster: Entity, action: CardBase, target: Array[Entity]):
-	self.caster = caster
-	self.action = action
-	self.target = target
+func _init(caster: Entity, action: CardBase, target_list: Array[Entity]):
```

Comment:
I believe underscored variables are in most cases used for instance variables, so it might be a bit confusing - we could also rename the function ones to be slightly different but I didn't have an idea at the time so I opted for adding underscores to the class variables

Reply:
In GDScript convention, variables using an underscore are "private" and are not meant to be accessed from outside the scope of the current script, just called by this script and children of this script
Thus it would be more logical to use `¢aster`, `ŧarget` and `target_list` for the class variables and `_caster`,`_target` and `_target_list` for the `_init` variable
As you can see, `_ready` functions and `_init` functions are a good example of that. They use an underscore because you don't call them directly (`_init` gets called by new() and `_ready` gets called when the tree is building the nodes)

Reply:
Yes, I'm saying the same thing as you are - it's just that the underscore in name is overall usually reserved for instance variables or methods that are private rather then for function arguments. Even the examples you mentioned are class functions (_init, _ready). Although I'm not sure how closely godot follows it, but from what I've seen it does seem to be the case

Overall it's just a convention though and not a big problem as the init should be pretty obvious anyway, so I swapped them around 👍

Reply:
well for now you have class variables with an underscore, (meaning private), and function parameters without an underscore (meaning public)
I would expect it to be other way around